src/server/routes/editor.tsx

```diff
@@ -531,6 +531,16 @@ router.get('/:id/achievements', async (req, res, next) => {
 		.where('editor_id', userId)
 		.fetchAll({require: false});
 
+	const achievementCol = await new AchievementUnlock()
```

Comment:
While `await`ing here works, it's a departure from the rest of this function where we create promises and then `await Promise.all(…` all the promises.

I'm not attached to Promise.all, especially if there is no strong reason why we need to await all promises at the same time.

I would suggest either following the current paradigm, or refactoring the promises so that they all get `await`-ed straight away instead of using Promise.all.
It will probably mean more try-catch blocks, for example to catch `getIdEditorJSONPromise` failures

Either way works for me.

Reply:
Fixed